### DISCUSSION TITLE
v4 alpha1

### DISCUSSION THREAD

**Discussion by RussKie** [gitextensions/gitextensions#10131] [General]
### Release Notes Highlights

* The app now requires .NET 6
* Improved repository load time considerably (#9243, #9735, #9864, etc) as well as other performance tweaks.
* Use native WSL Git executables for `\\wsl$` shares to speedup handling (#9702)
* Use the standard `FormBrowse` for file history and blame (#9445)
* BASE diff: Icons for unique changes (#9720)

### The following features and options have been removed:
* PuTTY binaries are no longer bundled with Git Extentions, please install PuTTY separately and change the path in settings.
* Theming is not supported - v3.x relied on EasyHook library to facilitate hooks into Windows API necessary for theming. However, .NET 6 does not support remoting and multiple AppDomain required for those hooks. To date no viable alternative has been found. https://github.com/gitextensions/gitextensions/pull/9191
* Removed TFS integration (https://github.com/gitextensions/gitextensions/pull/8522/commits/8190d0c0570af95ffbf9e168072d1b169eb99618)
* Removed JIRA integration (see: https://github.com/gitextensions/gitextensions/issues/9659)

<hr /><em>This discussion was created from the release <a href='https://github.com/gitextensions/gitextensions/releases/tag/v4-alpha1'>v4 alpha1</a>.</em>

**Comment by WhiteChairFromIkea** [comment]
Great news, but not having a dark theme is a no go for modern user 😕

**Reply by arkfen** [reply]
yeah, actually after testing alfa for a few days I did not find issues and did not know how I could help with feedback, but then I saw this request and remembered that myself anticipated the dark theme for a long time, so I hope you guys are rock and roll it soon as well!
great job in general by the way! thanks for all your hard work - really nice and handy GIF app, especially for GUI freaks like me ))

**Comment by borisdamevin** [comment]
I agree for dark mode, in 2022 this can't be missing and the `.msi` don't include .NET 6, we need to install it after.

**Comment by alastairtree** [comment]
Just helped a colleague with installing this and they had some issues. 

The installer had failed due to the missing dependency on .net 6 runtime and the .net6 desktop package. They googled the error, and installed both but (i suspect) installed an older version of both, which then crashed gitextentions silently on launch with an assembly manifest difference exception for the windows forms dll (identified from errors in Application logs in  event viewer). They tried reinstalling everything but it did not fix it.

The solution was to install the latest version - .NET Desktop Runtime 6.0.8 (also ended up installing the SDK too by accident, not sure if this was actually required).

The runtime is available from https://dotnet.microsoft.com/en-us/download/dotnet/6.0

In summary, I suspect gitex requires a particular minimum version of the .net6 desktop, and not just any 6.0 version, and so maybe the installer needs to validate more specifically. Hope this helps someone else.

**Reply by gerhardol** [reply]
GE uses DotnetRuntimeBootstrapper to avoid a manual .NET6 install or to include the about 50MB framework in the app.
I see no way to require a version, but asked a question about it.
https://github.com/Tyrrrz/DotnetRuntimeBootstrapper/discussions/33
